pkg/genericcli/context.go

```diff
@@ -0,0 +1,840 @@
+			deletedCtx = ctx
+		}
+		return ctx.Name == name
+	})
+
+	return deletedCtx
+}
+
+func (cs *contexts) getByName(name string) (*Context, bool) {
+	for _, context := range cs.Contexts {
+		if context.Name == name {
+			return context, true
+		}
+	}
+	return nil, false
+}
+
+func contextSorter() *multisort.Sorter[*Context] {
+	return multisort.New(multisort.FieldMap[*Context]{
+		sortKeyName: func(a, b *Context, descending bool) multisort.CompareResult {
+			return multisort.Compare(a.Name, b.Name, descending)
+		},
+		sortKeyAPIURL: func(a, b *Context, descending bool) multisort.CompareResult {
+			urlA := pointer.SafeDeref(a.APIURL)
+			urlB := pointer.SafeDeref(b.APIURL)
+			return multisort.Compare(urlA, urlB, descending)
+		},
+		sortKeyDefaultProject: func(a, b *Context, descending bool) multisort.CompareResult {
+			return multisort.Compare(a.DefaultProject, b.DefaultProject, descending)
+		},
+		sortKeyTimeout: func(a, b *Context, descending bool) multisort.CompareResult {
+			timeoutA := pointer.SafeDeref(a.Timeout)
+			timeoutB := pointer.SafeDeref(b.Timeout)
+			return multisort.Compare(timeoutA, timeoutB, descending)
+		},
+		sortKeyProvider: func(a, b *Context, descending bool) multisort.CompareResult {
+			return multisort.Compare(a.Provider, b.Provider, descending)
+		},
+	}, multisort.Keys{{ID: sortKeyName}})
+}
+
+func (c *ContextManager) GetContextCurrentOrDefault() *Context {
+	ctxs, err := c.getContexts()
+	if err != nil {
+		return defaultCtx()
+	}
+	ctx, ok := ctxs.getByName(ctxs.CurrentContext)
+	if !ok {
+		return defaultCtx()
+	}
+	// TODO deep copy?
+	// return ctx.deepCopy()
+	return ctx
+}
+
+func defaultCtx() *Context {
+	return &Context{
+		Name:     DefaultContextName,
+		APIURL:   pointer.PointerOrNil(viper.GetString(keyAPIURL)),
+		APIToken: viper.GetString(keyAPIToken),
+	}
+}
+
+func (c *ContextManager) GetCurrentContext() (*Context, error) {
+	ctxList, err := c.List()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, ctx := range ctxList {
+		if ctx.IsCurrent {
+			return ctx, nil
+		}
+	}
+	return nil, nil
+}
+
+func DefaultContext(c *ContextManager) (*Context, error) {
+	ctxs, err := c.getContexts()
+	if err != nil {
+		return nil, err
+	}
+
+	ctxName := ctxs.CurrentContext
+	if viper.IsSet(keyContextName) {
+		ctxName = viper.GetString(keyContextName)
+	}
+
+	ctx, ok := ctxs.getByName(ctxName)
+	if ok {
+		return ctx, nil
+	}
+
+	defaultCtx := c.GetContextCurrentOrDefault()
+	defaultCtx.Name = "default"
+
+	if ctxCurrent, ok := ctxs.getByName(ctxs.CurrentContext); ok {
+		ctxCurrent.IsCurrent = false
+	}
+	defaultCtx.IsCurrent = true
+	ctxs.PreviousContext, ctxs.CurrentContext = ctxs.CurrentContext, defaultCtx.Name
+	ctxs.Contexts = append(ctxs.Contexts, defaultCtx)
+
+	err = c.writeContexts(ctxs)
+	if err != nil {
+		return nil, fmt.Errorf(errMsgCannotWriteContexts, err)
+	}
+
+	ctx = defaultCtx
+
+	return ctx, nil
+}
+
+func ContextTable(data any, wide bool) ([]string, [][]string, error) {
```

Comment:
This is unused. Is this intentional? If it is used by users, move it up and document it.